src/passkey_child/passkey_child_devices.c

```diff
@@ -55,32 +55,173 @@ list_devices(fido_dev_info_t *dev_list, size_t *dev_number)
 }
 
 errno_t
-select_device(fido_dev_info_t *dev_list, size_t dev_index, fido_dev_t *dev)
+select_device(fido_dev_info_t *dev_list, size_t dev_list_len,
+              fido_assert_t *assert, fido_dev_t **_dev)
 {
+    fido_dev_t *dev = NULL;
     const char *path;
     const fido_dev_info_t *di = NULL;
     errno_t ret;
 
-    di = fido_dev_info_ptr(dev_list, dev_index);
-    if (di == NULL) {
-        DEBUG(SSSDBG_OP_FAILURE, "fido_dev_info_ptr failed.\n");
-        ret = ENOMEM;
+    if (dev_list_len == 0) {
+        DEBUG(SSSDBG_OP_FAILURE, "No device found. Aborting.\n");
+        ret = ENOENT;
         goto done;
+    } else if (dev_list_len == 1) {
```

Comment:
If there is only one device, no pre-flights are performed. This means that if there's a single authenticator connected to the host which does not contain any known credentials, there will be at least one assertion request fired (two if the device has both built-in UV and PIN configured). While not a blocker (the two assertion requests will both fail with `FIDO_ERR_NO_CREDENTIALS`) it is a distinct difference in behavior from the case where multiple authenticators are plugged in and they are all checked whether they contain any known credential before firing off the actual assertion request.

Reply:
As this function is used by the registration and the authentication problem I had to change it a little bit to offer the same behaviour.

Reply:
LGTM! :+1: